kalite/topic_tools/content_models.py

```diff
@@ -416,19 +427,15 @@ def recurse_availability_up_tree(node, available):
                     recurse_availability_up_tree(parent, available)
 
 
-            for path, update in updates_dict.iteritems():
+            for path, update in content_models:
                 if update:
                     # We have duplicates in the topic tree, make sure the stamping happens to all of them.
-                    items = Item.select().where(Item.path == path)
+                    
+                    items = Item.select().where((Item.path == path) & (Item.kind != "Topic"))
+                    Item.update("available", True).where((Item.path == path) & (Item.kind != "Topic"))
                     for item in items:
```

Comment:
It turns out we don't need this for loop anymore; the IDs passed into this function are already updated through the `update_content_availability` call. Instead we need to make sure that each unique path is passed into the `recurse_availability_up_tree` function.